parcels/field.py

```diff
@@ -485,10 +485,20 @@ def from_xarray(cls, da, name, dimensions, mesh='spherical', allow_time_extrapol
         interp_method = kwargs.pop('interp_method', 'linear')
 
         time = da[dimensions['time']].values if 'time' in dimensions else np.array([0])
-        depth = da[dimensions['depth']].values if 'depth' in dimensions else np.array([0])
         lon = da[dimensions['lon']].values
         lat = da[dimensions['lat']].values
 
+        if 'depth' in dimensions:
+            if dimensions['depth'] == 'not_yet_set':
+                _lon_lat_time = [dimensions[d] for d in ['lon', 'lat', 'time']]
+                _dim_depth = [d for d in da.dims if d not in _lon_lat_time][0]
+                depth = np.empty((0, len(da[_dim_depth].values), len(lat), len(lon)))
```

Comment:
well, this is nice but overly simplistic and not generalized sufficiently, I think.

Say you have a NetCDF file (or an xarray structure with a similar structure) with the following 'dimensions' (i.e. 'variables' in NetCDF):

- lon
- lat
- time
- level (of detail)
- depth_index
- depth (== `not_yet_set`)

(in that order)
Now, this code would not identify `depth`, but `_dim_depth` would be set to `level`, and the process would break. Sure, we are using the `from_xarray(...)` function here, but the xarray can be arranged in the same format as the file. Auto-parsing this dimensions (from a NetCDF) and actually identifying them is quite intricate - have a look into `parcels/fieldfilebuffer.py`, class `DaskFileBuffer`, function `_is_dimension_in_dataset(self, ...)`, line 401.

For the chosen (i.e. picked) test cases, it works - but it can be made to break fairly easy this way. Just saying as a comment.

Reply:
apologizes, I've had to drop this momentarily, please give me a couple of days to get back to you (on that and on the xarray lazyness issue which I've started to investigate).